src/web/app/components/extension-confirm-modal/extension-confirm-modal.component.spec.ts

```diff
@@ -90,6 +132,254 @@ describe('ExtensionConfirmModalComponent', () => {
 
   it('should snap with the extended students and instructors', () => {
     component.studentData = [studentModel1, studentModel2, studentModel3];
+    component.instructorData = [instructorModel2, instructorModel1, instructorModel3];
+    component.extensionTimestamp = testFeedbackSession.submissionEndTimestamp;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('use ngOnInit to initialise', () => {
+    component.selectedStudents = [studentModel3];
+    component.selectedInstructors = [instructorModel3];
+    component.ngOnInit();
+    component.extensionTimestamp = testFeedbackSession.submissionEndTimestamp;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
```

Comment:
let's also expect that `this.setStudentTableData();` and `this.setInstructorTableData();` are called

Reply:
we should also add a test where the length is 0, and expect that setStudentTableData and setInstructorTableData are not called

Reply:
Oh, that's good! I missed this possiblity